vars/cloudFoundryDeploy.groovy

```diff
@@ -461,8 +464,8 @@ def deployCfNative(config) {
         config.cloudFoundry.appName,
         config.deployOptions,
         config.cloudFoundry.manifest ? "-f '${config.cloudFoundry.manifest}'" : null,
-        config.deployDockerImage ? "--docker-image ${config.deployDockerImage}" : null,
-        binding.hasVariable("dockerUsername") ? "--docker-username ${dockerUsername}}" : null,
+        config.deployDockerImage && config.deployType != 'blue-green' ? "--docker-image ${config.deployDockerImage}" : null,
```

Comment:
Is it generally not possible to use blue/green deployments in combination with Docker images? I cannot find anything explicit in the [CF documentation](https://docs.cloudfoundry.org/devguide/deploy-apps/blue-green.html).

Reply:
Yes, that's the reason for this change.
But it actually can do docker deployments if the docker image and username are defined in a CF Manifest https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#docker The plugin just does not support most of the CF push cli parameters like --docker-username and --docker-image